File: mall-website/src/api/product/category.ts
import request, { ApiResult } from '../request'

// 分类实体
export interface Category {
  id: number
  name: string
  parentId: number
  level: number
  icon?: string
  children?: Category[]
}

// 获取分类树
export const getCategoryTree = () => {
  return request.get<any, ApiResult<Category[]>>('/product/category/tree')
}

// 获取分类列表
export const getCategoryList = (params: { name?: string; level?: number; parentId?: number }) => {
  return request.get<any, ApiResult<Category[]>>('/product/category/list', { params })
}

File: mall-product/src/main/java/com/hj/mall/product/entity/Category.java
package com.hj.mall.product.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("pms_category")
public class Category {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private String name;

    private Long parentId;

    private Integer level;

    private String icon;

    private Integer sort;

    private Integer status;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}
